File: Examples/Outlook/VB/NetOffice COMAddin Examples/06 FormRegions/CustomFormRegion1.vb
Imports NetOffice
Imports Outlook = NetOffice.OutlookApi
Imports NetOffice.OutlookApi.Tools
Imports MSForms = NetOffice.MSFormsApi
Imports NetOffice.OutlookApi.Tools.Contribution

Public Class CustomFormRegion1
    Inherits OpenFormRegion

    Private textBox1 As Outlook.OlkTextBox
    Private commandButton1 As Outlook.OlkCommandButton

    Public Sub New(formRegion As Outlook.FormRegion)

        MyBase.New(formRegion)

        Dim form As MSForms.UserForm = formRegion.Form
        textBox1 = form.Controls("TextBox1").To(Of Outlook.OlkTextBox)()
        commandButton1 = form.Controls("CommandButton1").To(Of Outlook.OlkCommandButton)()

        If Not IsNothing(commandButton1) Then
            Dim handler As Outlook.OlkCommandButton_ClickEventHandler = AddressOf Me.CommandButton1_ClickEvent
            AddHandler commandButton1.ClickEvent, handler
        End If

    End Sub

    Private Sub CommandButton1_ClickEvent()

        If Not IsNothing(textBox1) Then
            OutlookDialogUtils.ShowMessageBox(textBox1.Text, "Outlook06AddinVB4")
        End If

    End Sub

End Class

File: Examples/Outlook/C#/NetOffice COMAddin Examples/06 FormRegions/CustomFormRegion1.cs
using System;
using NetOffice;
using Outlook = NetOffice.OutlookApi;
using NetOffice.OutlookApi.Tools;
using MSForms = NetOffice.MSFormsApi;
using NetOffice.OutlookApi.Tools.Contribution;
using NetOffice.Extensions.Conversion;

namespace Outlook06AddinCS4
{
    public class CustomFormRegion1 : OpenFormRegion
    {
        private Outlook.OlkTextBox _textBox1;
        private Outlook.OlkCommandButton _commandButton1;

        public CustomFormRegion1(Outlook.FormRegion formRegion) : base(formRegion)
        {
            MSForms.UserForm form = formRegion.Form as MSForms.UserForm;
            _textBox1 = form.Controls["TextBox1"].To<Outlook.OlkTextBox>();
            _commandButton1 = form.Controls["CommandButton1"].To<Outlook.OlkCommandButton>();
            
            if(null != _commandButton1)     
                _commandButton1.ClickEvent += CommandButton1_ClickEvent;
        }

        private void CommandButton1_ClickEvent()
        {
            if(null != _textBox1)
                OutlookDialogUtils.ShowMessageBox(_textBox1.Text, "Outlook06AddinCS4");
        }
    }
}
